gufe/protocols/protocol.py

```diff
@@ -170,7 +171,7 @@ def create(
         *,
         stateA: ChemicalSystem,
         stateB: ChemicalSystem,
-        mapping: Union[dict[str, ComponentMapping], None],
+        mapping: Optional[Union[ComponentMapping, list[ComponentMapping], None]],
```

Comment:
```suggestion
        mapping: Optional[Union[ComponentMapping, list[ComponentMapping]]],
```
Doesn't Optional by default define a `None` case?

Reply:
https://docs.python.org/3/library/typing.html#typing.Optional